lab_1_classify_by_unigrams/main.py

```diff
@@ -14,6 +22,17 @@ def tokenize(text: str) -> list[str] | None:
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
+    if type(tokens) != list or tokens == [] or not all(isinstance(element, str) for element in tokens):
```

Comment:
`isinstance` is all you need
if you want to check a container for emptity - use len or BETTER us truthy/falsy : https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false